fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergeAggregate.java

```diff
@@ -0,0 +1,211 @@
+                return false;
+            }
+            if (outerFunc.isDistinct() && !sameGroupBy) {
+                return false;
+            }
+            // not support outerAggFunc: sum(a+1),sum(a+b)
+            if (!(outerFunc.child(0) instanceof SlotReference)) {
+                return false;
+            }
+            ExprId childExprId = ((SlotReference) outerFunc.child(0)).getExprId();
+            if (innerAggExprIdToAggFunc.containsKey(childExprId)) {
+                AggregateFunction innerFunc = innerAggExprIdToAggFunc.get(childExprId);
+                if (innerFunc.isDistinct() && !sameGroupBy) {
+                    return false;
+                }
+                // support sum(sum),min(min),max(max),any_value(any_value),sum(count)
+                // sum(count) -> count() need outerAgg having group by keys (reason: nullable)
+                if (!(outerFunc.getName().equals("sum") && innerFunc.getName().equals("count")
+                        && !outerAgg.getGroupByExpressions().isEmpty())
+                        && !innerFunc.getName().equals(outerFunc.getName())) {
+                    return false;
+                }
+            } else {
+                // select a, max(b), min(b), any_value(b) from (select a,b from t1 group by a, b) group by a;
+                // equals select a, max(b), min(b), any_value(b) from t1 group by a;
+                if (!outerFunc.getName().equals("max")
+                        && !outerFunc.getName().equals("min")
+                        && !outerFunc.getName().equals("any_value")) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    private boolean canMergeAggregateWithoutProject(LogicalAggregate<LogicalAggregate<Plan>> outerAgg) {
+        LogicalAggregate<Plan> innerAgg = outerAgg.child();
+        if (!new HashSet<>(innerAgg.getGroupByExpressions()).containsAll(outerAgg.getGroupByExpressions())) {
+            return false;
+        }
+        boolean sameGroupBy = (innerAgg.getGroupByExpressions().size() == outerAgg.getGroupByExpressions().size());
+
+        return commonCheck(outerAgg, innerAgg, sameGroupBy);
+    }
+
+    private boolean canMergeAggregateWithProject(LogicalAggregate<LogicalProject<LogicalAggregate<Plan>>> outerAgg) {
+        LogicalProject<LogicalAggregate<Plan>> project = outerAgg.child();
+        LogicalAggregate<Plan> innerAgg = project.child();
+
+        List<Expression> outerAggGroupByKeys = PlanUtils.replaceExpressionByProjections(project.getProjects(),
+                outerAgg.getGroupByExpressions());
+        if (!new HashSet<>(innerAgg.getGroupByExpressions()).containsAll(outerAggGroupByKeys)) {
+            return false;
+        }
+        // project cannot have expressions like a+1
+        if (ExpressionUtils.anyMatch(project.getProjects(),
+                expr -> !(expr instanceof SlotReference) && !(expr instanceof Alias))) {
```

Comment:
the code allows Alias( a + 1), this is different from comment. Please double check the expected behavior